src/main/scala/mesosphere/mesos/Constraints.scala

```diff
@@ -129,6 +128,43 @@ object Constraints {
     new ConstraintsChecker(tasks, offer, constraint).isMatch
 
   /**
+    * Select slaves based on given constraints.
+    * @param slaves the iterable sequence of slaves to select from.
+    * @param constraints the constraints that all slaves have to meet.
+    * @return the filtered list of slaves that meet all constraints.
+    */
+  def selectMatchingSlaves(slaves: Iterable[MesosSlaveData],
```

Comment:
Can we maybe reuse the `meetsConstraint` method above with an empty task list? Then we do not have to take care to keep this in sync. It's a bit weird with constraints like group_by, admitted, but with the empty task list it should have a similar effect.